devito/types/sparse.py

```diff
@@ -709,22 +694,28 @@ def _dist_scatter(self, data=None):
         comm = distributor.comm
         mpitype = MPI._typedict[np.dtype(self.dtype).char]
 
+        # Compute dist map only once
+        dmap = self._dist_datamap
+
         # Pack sparse data values so that they can be sent out via an Alltoallv
-        data = data[self._dist_scatter_mask]
+        data = data[self._dist_scatter_mask(dmap=dmap)]
```

Comment:
@FabioLuporini @rhodrin This is basically 90-95% of the cost because we index a numpy array with individual indices so it's quite inefficient. But we can't really get out of it since we can't guaranty that the indices will be consecutive so gonna be really hard to get better.

Reply:
Same below for coords indexing.